### ISSUE TITLE
Dynamic link title generation?

### ISSUE THREAD

**Issue by oliverguenther** [sammyhenningsson/hal_presenter#3]
We sometimes use properties of the resource for generation of links. For example if you link to a user's preferences, you might want to show the user name in the link.

As far as I've seen, hal_presenter doesn't let you do that as the title is a static property. You could probably extend the Link response to check for a hash being present, that would allow the block to return a hash itself. Something like the following:

```ruby

link :preferences do
  {
    href: preferences_path(resource),
    title: "Preferences for #{resource.name}" 
  }
end

```

**Comment by sammyhenningsson** [comment]
Sorry, I totally missed this issue. That's a great idea!

**Comment by sammyhenningsson** [comment]
Implemented in version 1.7.0

**Comment by oliverguenther** [comment]
Very cool, thanks!
